Bottom: 78e101229f45af0c26d7a3929802ce37ce51e9f8
Top:    2bd50c180d0a8c3d6f476336bb92049576b654cc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-29 16:54:31 +0100

Hook up debug drawing APIs for GTK and Lesstif GUIs


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 27d62c6..b5e28b6 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1226,3 +1226,17 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
 
   return pixmap;
 }
+
+HID *
+ghid_request_debug_draw (void)
+{
+}
+
+void
+ghid_flush_debug_draw (void)
+{
+}
+
+ghid_finish_debug_draw (void)
+{
+}
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index f5e2717..ac961db 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2157,6 +2157,10 @@ hid_gtk_init ()
   ghid_hid.drc_gui                  = &ghid_drc_gui,
   ghid_hid.edit_attributes          = ghid_attributes;
 
+  ghid_hid.request_debug_draw       = ghid_request_debug_draw;
+  ghid_hid.flush_debug_draw         = ghid_flush_debug_draw;
+  ghid_hid.finish_debug_draw        = ghid_finish_debug_draw;
+
   hid_register_hid (&ghid_hid);
 #include "gtk_lists.h"
 }
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 6cdeff2..d52d088 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -499,6 +499,9 @@ gboolean ghid_drawing_area_expose_cb (GtkWidget *, GdkEventExpose *,
 gboolean ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev);
 GdkPixmap *ghid_render_pixmap (int cx, int cy, double zoom,
                                int width, int height, int depth);
+HID *ghid_request_debug_draw (void);
+void ghid_flush_debug_draw (void);
+void ghid_finish_debug_draw (void);
 
 /* gtkhid-main.c */
 void ghid_pan_fixup (void);
